src/autopas/containers/verletClusterLists/traversals/ClusterFunctor.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file ClusterFunctor.h
+ * @author humig
+ * @date 12.08.19
+ */
+
+#pragma once
+
+#include "autopas/containers/verletClusterLists/Cluster.h"
+#include "autopas/options/DataLayoutOption.h"
+#include "autopas/utils/ExceptionHandler.h"
+
+namespace autopas::internal {
+/**
+ * Provides methods to traverse a single cluster and a pair of clusters.
+ *
+ * @tparam Particle The type of particle the clusters contain.
+ * @tparam PairwiseFunctor The type of the functor the ClusterFunctor should use.
+ * @tparam dataLayout The data layout to use.
+ * @tparam useNewton3 If newton 3 should be used or not.
+ * @tparam clusterSize The number of particles in each cluster.
+ */
+template <class Particle, class PairwiseFunctor, DataLayoutOption::Value dataLayout, bool useNewton3,
+          size_t clusterSize>
+class ClusterFunctor {
+ public:
+  /**
+   * Constructs a ClusterFunctor that uses the given functor internally.
+   * @param functor The functor to use internally.
+   */
+  explicit ClusterFunctor(PairwiseFunctor *functor) : _functor(functor) {}
+
+  /**
+   * Traverses pairs of all particles in the given cluster. Always uses newton 3 in the AoS data layout.
+   * @param cluster The cluster to traverse.
+   */
+  void traverseCluster(internal::Cluster<Particle, clusterSize> &cluster) {
+    if constexpr (dataLayout == DataLayoutOption::aos) {
+      for (size_t i = 0; i < clusterSize; i++) {
+        // Always use newton 3 for interactions within one cluster.
+        for (size_t j = i + 1; j < clusterSize; j++) {
+          _functor->AoSFunctor(cluster.at(i), cluster.at(j), true);
```

Comment:
especially in functions that are incredibly often called we really should consider using `[]` over `at()`. Is there a reason why `at()` should stay here? If no change it please.